src/mutation.cpp

```diff
@@ -654,10 +1052,11 @@ void player::power_mutations()
         const long ch = ctxt.get_raw_input().get_first_input();
         if (menu_mode == "reassigning") {
             menu_mode = "activating";
-            std::string mut_id;
-            for( const auto &key_pair : trait_keys ) {
-                if( key_pair.second == ch ) {
-                    mut_id = key_pair.first;
+            muttype_id mut_id;
+            // Find the selected mutation by key
+            for( const auto &mut : my_mutations ) {
```

Comment:
This loop should be in a function that return a pointer to the mutation (or null), like `bionic_by_invlet`. It's repeated several times.